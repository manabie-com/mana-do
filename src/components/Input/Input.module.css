.root {
  min-height: 36px;
  border: none;
  outline: none;
  padding: 0 12px;
  box-shadow: 2px 0 4px rgba(0,0,0, 0.2);
  border-radius: 4px;
}
.root:focus {
  box-shadow: 1px 0 9px rgba(0,0,0, 0.25);
}