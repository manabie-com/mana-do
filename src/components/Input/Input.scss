@import "../../assets/scss/variables";

.input {
  width: 100%;
  border: solid 1px rgba($color: #000000, $alpha: 0.1);
  border-radius: 8px;
  position: relative;
  max-width: 300px;
  height: 46px;
  &__lable {
    color: $primiry-color;
    position: absolute;
    top: -11px;
    font-size: 14px;
    background-color: white;
    left: 10px;
    text-transform: capitalize;
    font-weight: 500;
  }
  &__input {
    border: none;
    background-color: transparent;
    width: 100%;
    height: 100%;
    padding: 0 8px;
    font-size: 1rem;
    &:focus {
      outline: none;
    }
  }
  &__error {
    color: $error-color;
    font-size: 12px;
  }
}
