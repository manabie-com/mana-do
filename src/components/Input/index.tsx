import React from 'react';

interface inputProps {
    onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
    className?: string | undefined;
    placeholder?: string;
    value?: string;
}

const Input = (props: inputProps) => {
    const { onKeyDown, className, placeholder, onChange, value } = props
    return (
        <input
            value={value}
            className={`${className} Action__btn`}
            onKeyDown={onKeyDown}
            placeholder={placeholder}
            onChange={onChange}
            type="text"
        />
    )
}

export default Input