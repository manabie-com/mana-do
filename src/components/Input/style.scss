.custom-input{
  label{
    color: white;
    font-weight: bold;
  }
  input {
    width: 100%;
    min-height: 36px;
    border: none;
    outline: none;
    padding: 0 12px;
    box-shadow: 2px 0 4px rgba(0,0,0, 0.2);
    border-radius: 4px;
    &[type="checkbox"] {
      width: 24px;
      height: 24px;
      box-shadow: none;
      border: none;
      outline: none;
    }
    &[type="checkbox"]:focus {
      box-shadow: none;
      border: none;
      outline: none;
    }
    &:focus {
      box-shadow: 1px 0 9px rgba(0,0,0, 0.25);
    }
  }
}