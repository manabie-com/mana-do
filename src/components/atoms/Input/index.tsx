import React from 'react';
import styles from './styles.module.scss';

interface InputProps {
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void,
  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void,
  value?: string,
  name?: string,
  type?: string,
  placeholder?: string,
  className?: string,
  id?: string,
  innerRef?: any
}

const Input = (props: InputProps) => {
  const {
    onChange, value, placeholder, className, type, name, innerRef, onKeyDown, id
  } = props;
  return (
    <input
      ref={innerRef}
      type={type}
      name={name}
      defaultValue={value}
      onChange={onChange}
      onKeyDown={onKeyDown}
      className={`${styles.Input} ${className}`}
      id={id}
      placeholder={placeholder}
    />
  );
};

Input.defaultProps = {
  type: 'text',
  name: '',
  placeholder: '',
  className: '',
};

export default Input;
