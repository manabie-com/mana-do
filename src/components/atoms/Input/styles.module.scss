input[type=text], input[type=password] {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  box-sizing: border-box;
}
