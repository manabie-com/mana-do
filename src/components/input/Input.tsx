import React, { forwardRef, InputHTMLAttributes, memo } from "react";
import { InputType } from "../../models";
import "./styles.scss";

export interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {
  placeholder?: string;
  className?: string;
  disabled?: boolean;
  id?: string;
  name: string;
  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => Promise<void>;
}

const Input = forwardRef<HTMLInputElement, IInputProps>(
  ({ placeholder, disabled, onKeyDown, className, id, name, ...props }, ref) => {
    return (
      <input
        type={InputType.TEXT}
        disabled={disabled}
        ref={ref}
        placeholder={placeholder}
        onKeyDown={onKeyDown}
        className={`input-todo ${className ? className : ""}`}
        id={id || ""}
        name={name}
        data-testid="input"
        {...props}
      />
    );
  }
);

export default memo(Input);
