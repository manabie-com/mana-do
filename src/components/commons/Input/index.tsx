import cn from "classnames";
import React, { forwardRef } from "react";
import "./style.css";

interface InputProps {
  className?: string;
  disabled?: boolean;
  readOnly?: boolean;
  type?: string;
  name?: string;
  id?: string;
  value?: string | number;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  placeholder?: string;
  checked?: boolean;
}

const Input = forwardRef((props: InputProps, ref: any) => {
  const { value, className, disabled, ...rest } = props;
  const classOfRoot = cn("root-input", className);
  const classOfInputBase = cn(
    "input-custom",
    "input-size-md",
    disabled && "input-custom__disabled"
  );

  return (
    <div className={classOfRoot}>
      <input value={value} className={classOfInputBase} ref={ref} {...rest} />
    </div>
  );
});

export default Input;
