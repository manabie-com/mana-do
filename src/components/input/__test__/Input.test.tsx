import { fireEvent, render, RenderResult } from "@testing-library/react";
import React from "react";
import { getElementByTestId } from "utils/testUtils";
import Input, { IInputProps } from "../Input";

const onKeyDownAction = jest.fn();

const renderComponent = ({ ...props }: IInputProps): RenderResult => {
  return render(<Input {...props} />);
};

describe("<Input /> rendering", () => {
  it("should match snapshot", () => {
    const wrapper = renderComponent({ name: "todo-input" });
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should render correcly", () => {
    renderComponent({ name: "todo-input" });
    const element = getElementByTestId("input");
    expect(element).toBeInTheDocument();
  });

  it("should render a input with the default class", () => {
    renderComponent({ name: "todo-input" });
    const element = getElementByTestId("input");
    expect(element).toHaveClass("input-todo");
  });
});

describe("<Input /> interacting", () => {
  it("should render a input clickable", () => {
    renderComponent({ name: "todo-input" });
    const element = getElementByTestId("input");
    expect(element).not.toBeDisabled();
  });

  it("should handle onKeydown", () => {
    renderComponent({
      name: "todo-input",
      onKeyDown: onKeyDownAction,
    });
    const element = getElementByTestId("input");

    fireEvent.keyDown(element, {
      key: "Enter",
      code: "Enter",
      charCode: 13,
    });

    expect(onKeyDownAction).toHaveBeenCalled();
  });
});
