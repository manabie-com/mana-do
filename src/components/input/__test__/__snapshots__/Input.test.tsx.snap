// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> rendering should match snapshot 1`] = `
<div>
  <input
    class="input-todo "
    data-testid="input"
    id=""
    name="todo-input"
    type="text"
  />
</div>
`;
